src/cascadia/TerminalApp/AppLogic.cpp

```diff
@@ -321,7 +323,7 @@ namespace winrt::TerminalApp::implementation
         auto loadedRevoker{ dialog.Loaded(winrt::auto_revoke, themingLambda) }; // if it's not yet in the tree
 
         // Display the dialog.
-        co_await dialog.ShowAsync(Controls::ContentDialogPlacement::Popup);
+        return co_await dialog.ShowAsync(Controls::ContentDialogPlacement::Popup);

```

Comment:
Since we're just passing the action back through, can this be simplified down to `return dialog.Show`? Or is this fine/okay/better

Reply:
No, that would try to return `IAsyncOperation<IAsyncOperation<ContentDialogResult>>`.

The problem is that I don't know how to construct an `IAsyncOperation<ContentDialogResult>` myself with a `ContentDialogResult`. And because I didn't know, then I made the whole function a coroutine so that the compiler does it for me when I do `co_return ContentDialogResult::None`. But it also means that I have to "unwrap" the result of `dialog.ShowAsync` to let the compiler wrap it again in a `IAsyncOperation<ContentDialogResult>`.

I don't know if that's very clear. If you know JavaScript, I'm looking for something like `Promise.resolve` so I can get rid of the whole coroutine thing with the `co_return co_await` but I couldn't find it.

Reply:
Yeah, that's sorta what I was expecting. Thanks for the explanation.

I was hoping for something like "just return the promise object that this function returns".. but given that you have to wrap `None`, this is totally fine